pandas/core/strings.py

```diff
@@ -35,19 +35,32 @@
 _shared_docs = dict()
 
 
-def _get_array_list(arr, others):
+def _get_array_list(arr, others, align=True):
     from pandas.core.series import Series
 
-    if len(others) and isinstance(com._values_from_object(others)[0],
-                                  (list, np.ndarray, Series)):
+    if (len(arr) and isinstance(arr, Series)
+            and len(others) and isinstance(others, Series)):
+        if align is None:
+            align = False
+            warnings.warn("A future version of pandas will perform alignment "
+                          "when others is a series. To disable alignment (the "
+                          "previous behavior) and silence this warning, pass "
+                          "'align=False'. To enable alignment (the future "
+                          "behavior) and silence this warning, pass "
+                          "'align=True'", FutureWarning, stacklevel=4)
+        if align:
+            arr, others = arr.align(others, join='outer')
+        arrays = [list(arr), list(others)]
```

Comment:
I don't think the conversion to list here is needed (on line 60 they get converted to an array anyhow)

Reply:
This was part of the ``_get_array_list``-function as I found it - I didn't investigate how it interplays with ``str_cat`` (which is the only place it called from), so I left it as it was.